### PR TITLE
Add ASC proposal for `SPE`

### PR DESCRIPTION
**Pull request by VarunKanwar**
A proposal to add an `spe` instruction to represent solid phase extraction in Autoprotocol.

### CODE REVIEW

File: Proposed/asc047.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+#### **Proposal**
+We propose the addition of an `spe` instruction to Autoprotocol to represent the separation of a liquid sample into multiple eluted fractions via solid phase extraction.
+
+```
+{
+  "op": "spe",
+  /* Source aliquot with liquid matrix */
+  "object": Well,
+  /* Sorbent cartridge used to extract the target compound. Required */
+  "cartridge": String or ColumnType,
```
Comment (polarpine) [inline Proposed/asc047.md]: remove ColumnType

File: Proposed/asc047.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+   * whereas negative pressure is applied from below the cartridge as vacuum.
+   */
+  "pressure_mode": Enum("positive", "negative")
+  /* Containers used to collect eluates. Required */
+  "elution_containers": List(Container)
+  /*
+   * Parameters for conditioning the cartridge with one or more solvents.
+   * Optional
+   */
+  "conditioning": List(
```
Comment (polarpine) [inline Proposed/asc047.md]: use the same tense for all the mobile phase steps, eg. 
`condition`, `equilibrate`, `load_sample`, `rinse`, `elute`

File: Proposed/asc047.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  "object": Well,
+  /* Sorbent cartridge used to extract the target compound. Required */
+  "cartridge": String or ColumnType,
+  /*
+   * The direction of pressure applied to the cartridge to force liquid flow.
+   * Positive pressure is applied from above the cartridge as a pushing force,
+   * whereas negative pressure is applied from below the cartridge as vacuum.
+   */
+  "pressure_mode": Enum("positive", "negative")
+  /* Containers used to collect eluates. Required */
```
Comment (polarpine) [inline Proposed/asc047.md]: Add note that each container must have a corresponding `elute` parameter, e.g. if 3 containers are submitted as `elution_containers` there must be 3 dictionaries of `elute` params in the `elute` list.

File: Proposed/asc047.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+       */
+      "processing_time": Time,
+      /*
+       * Pressure in the column. This determines the rate at which the solvent
+       * is removed.
+       */
+      "flow_pressure": Pressure
+    }
+  ),
+  /* Parameters for equilibrating the cartridge with matrix-like solvents */
```
Comment (polarpine) [inline Proposed/asc047.md]: Optional?

File: Proposed/asc047.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+     * the cartridge, after `settle_time` has elapsed.
+     */
+    "processing_time": Time,
+    /*
+     * Pressure in the column. This determines the rate at which the matrix
+     * is removed.
+     */
+    "flow_pressure": Pressure
+  },
+  /* Parameters for eluting interferents */
```
Comment (polarpine) [inline Proposed/asc047.md]: Optional? change "eluting" to "rinse" in docstring

**Review by polarpine — CHANGES_REQUESTED** [review]

File: Proposed/asc047.md (hunk 3 above)
Comment (VarunKanwar) [reply Proposed/asc047.md]: Removed the top-level `elution_containers` field, and added a `destination_well` field to the `elute` group, which makes collection into specific containers less ambiguous.

**Review by polarpine — APPROVED** [review]

**Pull request merged**